genref/output/md/kubeadm-config.v1beta3.md

```diff
@@ -0,0 +1,1459 @@
+  kind: KubeProxyConfiguration
+     ...
+  ```
+
+The KubeProxyConfiguration type should be used to change the configuration passed to kube-proxy instances deployed
+in the cluster. If this object is not provided or provided only partially, kubeadm applies defaults.
+
+See https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/ or https://godoc.org/k8s.io/kube-proxy/config/v1alpha1#KubeProxyConfiguration
+for kube proxy official documentation.
+
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+...
+```
+
+The KubeletConfiguration type should be used to change the configurations that will be passed to all kubelet instances
+deployed in the cluster. If this object is not provided or provided only partially, kubeadm applies defaults.
+
+See https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/ or https://godoc.org/k8s.io/kubelet/config/v1beta1#KubeletConfiguration
+for kubelet official documentation.
+
+Here is a fully populated example of a single YAML file containing multiple
+configuration types to be used during a `kubeadm init` run.
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1beta3
+kind: InitConfiguration
+bootstrapTokens:
+  - token: "9a08jv.c0izixklcxtmnze7"
+    description: "kubeadm bootstrap token"
+    ttl: "24h"
+  - token: "783bde.3f89s0fje9f38fhf"
+    description: "another bootstrap token"
+    usages:
+      - authentication
+      - signing
+    groups:
+      - system:bootstrappers:kubeadm:default-node-token
+nodeRegistration:
+	 name: "ec2-10-100-0-1"
+	 criSocket: "/var/run/dockershim.sock"
+	 taints:
+	   - key: "kubeadmNode"
+	     value: "master"
+	     effect: "NoSchedule"
+	 kubeletExtraArgs:
+	   v: 4
+	 ignorePreflightErrors:
+	   - IsPrivilegedUser
+localAPIEndpoint:
+	 advertiseAddress: "10.100.0.1"
+	 bindPort: 6443
+certificateKey: "e6a2eb8581237ab72a4f494f30285ec12a9694d750b9785706a83bfcbbbd2204"
+---
+apiVersion: kubeadm.k8s.io/v1beta3
+kind: ClusterConfiguration
+etcd:
+  # one of local or external
+  local:
+	   imageRepository: "k8s.gcr.io"
+	   imageTag: "3.2.24"
+	   dataDir: "/var/lib/etcd"
+	   extraArgs:
+	     listen-client-urls: "http://10.100.0.1:2379"
+	   serverCertSANs:
+	     -  "ec2-10-100-0-1.compute-1.amazonaws.com"
+	   peerCertSANs:
+	     - "10.100.0.1"
+	 # external:
+	   # endpoints:
+	   # - "10.100.0.1:2379"
+	   # - "10.100.0.2:2379"
+	   # caFile: "/etcd/kubernetes/pki/etcd/etcd-ca.crt"
+	   # certFile: "/etcd/kubernetes/pki/etcd/etcd.crt"
+	   # keyFile: "/etcd/kubernetes/pki/etcd/etcd.key"
+	networking:
+	  serviceSubnet: "10.96.0.0/12"
+	  podSubnet: "10.100.0.1/24"
+	  dnsDomain: "cluster.local"
+	kubernetesVersion: "v1.12.0"
+	controlPlaneEndpoint: "10.100.0.1:6443"
+	apiServer:
+	  extraArgs:
+	    authorization-mode: "Node,RBAC"
+	  extraVolumes:
+	  - name: "some-volume"
+	    hostPath: "/etc/some-path"
+	    mountPath: "/etc/some-pod-path"
+	    readOnly: false
+	    pathType: File
+	  certSANs:
```

Comment:
Do you want to remove these tabs?

Reply:
oh ??? These are from the source code. Will investigate and remove them.